Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Connecting to Azure account..."
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    Write-Host "Current context has been set to below details: " -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+    Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+    Write-Host "Account Name: [$($context.Account.Id)]"
+    Write-Host "Account Type: [$($context.Account.Type)]"
+    Write-Host $([Constants]::SingleDashLine)
+        
+    Write-Host "***To add AAD Auth extension on Linux VMSS(s) in a Subscription, Contributor or higher privileges on the VMSS(s) are required.***" -ForegroundColor $([Constants]::MessageType.Warning)
+   
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all Virtual Machine Scale Set(s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list to store resource details.
+    $VMSSDetails = @()
+
+    #Separated Linux VMSS(s)
+    $NonCompliantVMSSDetails = @()
+
+    $reqExtPublisher = "Microsoft.Azure.ActiveDirectory"
+    $reqExtensionType = "AADSSHLoginForLinux"
+    $reqExtensionName = "AADSSHLoginForLinux"
+
+    # No file path provided as input to the script. Fetch all Virtual Machine Scale Set(s) in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        try
+        {
+            Write-Host "Fetching all Virtual Machine Scale Set(s) in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+            # Get all Virtual Machine Scale Set(s) in a Subscription
+            $VMSSDetails =  Get-AzVmss -ErrorAction Stop
+
+            # Seperating required properties
+            $VMSSDetails = $VMSSDetails | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                    @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                    @{N='ResourceName';E={$_.Name}},
+                                                                    @{N='OrchestrationMode';E={$_.OrchestrationMode}},
+                                                                    @{N='OsType';E={$_.VirtualMachineProfile.StorageProfile.OsDisk.OsType}},
+                                                                    @{N='Extensions';E={$_.VirtualMachineProfile.ExtensionProfile.Extensions}}
+                                 
```

Comment:
we should add some messages like how many VMSS found to notify users about that process 'this Fetching all Virtual Machine Scale Set(s) in Subscription:' ended.

Reply:
Added message as "Found [$($VMSSDetails.count)] Virtual Machine Scale Set(s) in the subscription..." and then new message that we are searching for non compliant linux based VMSS